### PR TITLE
feat(site): add descriptions for each auth method to the selection menu

### PR DESCRIPTION
**Pull request by aslilac**
Screenshot 2023-08-22 at 11 57 30 AM

Mostly motivated by the fact that just saying "None" was really cryptic and confusing.

### CODE REVIEW

File: site/src/components/CreateUserForm/CreateUserForm.tsx [hunk 1]
```diff
@@ -172,12 +171,41 @@ export const CreateUserForm: FC<
...
+            }}
           >
             {methods}
           </TextField>
+          <TextField
+            {...getFieldHelpers(
+              "password",
+              form.values.login_type === "password"
+                ? ""
+                : "No password required for this login type",
```
Comment (BrunoQuaresma) [inline site/src/components/CreateUserForm/CreateUserForm.tsx:187]: If the password is not required, why not hide this field?

Comment (aslilac) [reply site/src/components/CreateUserForm/CreateUserForm.tsx:187]: jitter, mostly. on some level, seeing another field after the auth method prepares the user for "there is another potential step" even if it's disabled. if it's entirely hidden, they might just jump straight to the submit button.

**Review by BrunoQuaresma — COMMENTED** [review]
Nit: I would do this a bit differently. 

Maybe we should make the `authMethods` required and make the parent display the loading state if it is not loaded since this field is required. So we could do something like:
```tsx
const availableMethods = authMethods.filter(method => method.enabled).concat("none")

<TextField>
  {availableMethods.map(method => {
    const language = authMethodLanguage[method.id]
    return (
      <MenuItem key={value} id={"item-" + value} value={value}>
        <Stack spacing={0} maxWidth={400}>
          {language.displayName}
          <span className={styles.labelDescription}>
            {language.description}
          </span>
        </Stack>
      </MenuItem>
    )
  })}
</TextField>
```

**Review by BrunoQuaresma — APPROVED** [review]

**Comment by aslilac** [comment]
that's probably nicer than doing the weird not-quite-a-component stuff. I'll do a quick refactor.

**Pull request merged**